pkg/kubelet/kubelet.go

```diff
@@ -1223,7 +1207,18 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, runningPod kubecont
 		return err
 	}
 
-	containerChanges, err := kl.computePodContainerChanges(pod, runningPod)
+	podStatus, err := kl.generatePodStatus(pod)
```

Comment:
I'd say to swap this with makePodDataDir() so the status is as updated as possible (still a race, but smaller).